whois-commons/src/main/resources/nrtm_schema.sql

```diff
@@ -0,0 +1,58 @@
+/*!40101 SET @OLD_CHARACTER_SET_CLIENT = @@CHARACTER_SET_CLIENT */;
+/*!40101 SET @OLD_CHARACTER_SET_RESULTS = @@CHARACTER_SET_RESULTS */;
+/*!40101 SET @OLD_COLLATION_CONNECTION = @@COLLATION_CONNECTION */;
+/*!40101 SET NAMES utf8 */;
+/*!40103 SET @OLD_TIME_ZONE = @@TIME_ZONE */;
+/*!40103 SET TIME_ZONE = '+00:00' */;
+/*!40014 SET @OLD_UNIQUE_CHECKS = @@UNIQUE_CHECKS, UNIQUE_CHECKS = 0 */;
+/*!40014 SET @OLD_FOREIGN_KEY_CHECKS = @@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS = 0 */;
+/*!40101 SET @OLD_SQL_MODE = @@SQL_MODE, SQL_MODE = 'NO_AUTO_VALUE_ON_ZERO' */;
+/*!40111 SET @OLD_SQL_NOTES = @@SQL_NOTES, SQL_NOTES = 0 */;
+
+DROP TABLE IF EXISTS `source`;
+create table `source`
+(
+    `id`   int(11) unsigned NOT NULL AUTO_INCREMENT,
```

Comment:
Do you need an `id` column if the `name` is unique?

Reply:
maybe here we can use the name. I understand the point of Paul, to use Id always to identify the row instead of another column (doesn't matter if that column is unique or not), I guess that this is just different point of view.

Reply:
yes that's really it. i'm not making any assumptions about uniqueness coz it's not 100% designed yet. when we've got a working implementation it might turn out that this could be optimized. even then, thinking about it purely as an abstract data source, it's almost always a good idea to have a dedicated unique identifier rather than overloading the role of the 'name'  column to be both a business code and technically unique, since then it has mixed responsibility, and introduces a constraint you might not always want.